src/pages/Account/Components/CodeSnippet.tsx

```diff
@@ -29,9 +32,18 @@ function useStartingLineNumber(sourceCode?: string) {
 
 function ExpandCode({sourceCode}: {sourceCode: string | undefined}) {
   const theme = useTheme();
+  const {selectedModuleName} = useParams();
+  const {account} = useWallet();
+  const [state] = useGlobalState();
   const [isModalOpen, setIsModalOpen] = useState(false);
 
   const handleOpenModal = () => {
+    Statsig.logEvent("expand_button_clicked", selectedModuleName, {
+      stable_id: Statsig.getStableID(),
+      wallet_address: account?.address ?? "",
+      network_type: state.network_name,
+    });
```

Comment:
can we turn this into a hook so that we do `useWallet` and `useGlobalState` inside the hook? to clean up code

Reply:
updated.